packages/core/src/browser/view-container.ts

```diff
@@ -276,22 +351,29 @@ export class ViewContainer extends BaseWidget implements StatefulWidget, Applica
 
     protected readonly toRemoveWidgets = new Map<string, DisposableCollection>();
 
-    addWidget(widget: Widget, options?: ViewContainer.Factory.WidgetOptions): Disposable {
+    createPartId(widget: Widget): string {
+        const description = this.widgetManager.getDescription(widget);
+        return widget.id || JSON.stringify(description);
```

Comment:
@colin-grant-work the `id` was created that way (with the `|| description`) before this PR changes. I have just moved it into a function because I use it in 2 places...